src/main/groovy/org/pih/warehouse/inventory/CycleCountTransactionCreator.groovy

```diff
@@ -0,0 +1,128 @@
+package org.pih.warehouse.inventory
+
+import grails.validation.ValidationException
+import org.springframework.stereotype.Component
+
+import org.pih.warehouse.core.Constants
+import org.pih.warehouse.product.Product
+
+/**
+ * Component responsible for creating and persisting the Transactions that result from a cycle count.
+ */
+@Component
+class CycleCountTransactionCreator {
+
+    ProductAvailabilityService productAvailabilityService
+    ProductInventorySnapshotTaker productInventorySnapshotTaker
+    TransactionIdentifierService transactionIdentifierService
+
+    /**
+     * Given a completed cycle count, will create and persist the resulting product inventory snapshot transaction,
+     * as well as the credit/debit adjustment transaction if there are discrepancies.
+     *
+     * @param itemQuantityOnHandIsUpToDate
+     */
+    List<Transaction> createTransactions(CycleCount cycleCount, boolean itemQuantityOnHandIsUpToDate=false) {
```

Comment:
When will `itemQuantityOnHandIsUpToDate` be passed as a `true`? Will it be a part of a request?